### DISCUSSION TITLE
Native application settings as a plugin

### DISCUSSION THREAD

**Discussion by yacut** [ionic-team/capacitor#3408] [Ideas]
It would be helpful to make native application settings with a capacitor plugin. e.g. preferred language, app version, about link

<img width="187" alt="iOS settings" src="https://user-images.githubusercontent.com/4711633/89916546-07f39e00-dbf8-11ea-96ab-cba752e74ac1.png">

**Comment by ikeith** [comment]
Exposing an iOS application's settings to the System Settings app (as shown in your screenshot) is not done with native code and therefore would not require a plugin. You define it by adding `.plist` files in a [Settings Bundle](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/UserDefaults/Preferences/Preferences.html) to the native project. Those values are then stored in the application's [UserDefaults](https://developer.apple.com/documentation/foundation/userdefaults) which can be accessed with the existing [Storage plugin](https://capacitorjs.com/docs/apis/storage).

**Reply by yacut** [reply]
Thank you for the answer. I will try it.

**Reply by 8eecf0d2** [reply]
To be clear, you define it by adding a .plist but the Storage plugin will only read keys which have the correct prefix (_cap_ by default.)
